cmake/dts.cmake

```diff
@@ -88,6 +88,63 @@ if(CONFIG_HAS_DTS)
   # preprocessor, and it seems to be including all kinds of
   # directories with who-knows how many header files.
 
+  # Run the C preprocessor on an empty C source file that has one or
+  # more DTS source files -include'd into it to create the
+  # intermediary file *.dts.pre.tmp
+  execute_process(
+    COMMAND ${CMAKE_C_COMPILER}
+    -x assembler-with-cpp
+    -nostdinc
+    -isystem ${DTS_APP_INCLUDE}
+    -isystem ${ZEPHYR_BASE}/include
+    -isystem ${ZEPHYR_BASE}/dts/${ARCH}
+    -isystem ${ZEPHYR_BASE}/dts
+    ${DTC_INCLUDE_FLAG_FOR_DTS}  # include the DTS source and overlays
+    -I${ZEPHYR_BASE}/dts/common
+    ${NOSYSDEF_CFLAG}
+    -D__DTS__
+    -P
+    -E ${ZEPHYR_BASE}/misc/empty_file.c
+    -o ${BOARD}.dts.pre.tmp
```

Comment:
This is the same file name as is used on the second pass.

It is not permitted to modify files in the build directory.

We need to use a different filename here.